src/sentinel.rs

```diff
@@ -208,4 +241,133 @@ mod test {
         value : usize
     }
 
+    impl TestClaim {
+        pub fn serialise(&self) -> SerialisedClaim {
+            self.value.to_string().into_bytes()
+        }
+    }
+
+#[test]
+    fn one_request_and_one_key() {
+        let mut name_key_pairs = Vec::new();
+        let mut sentinel: Sentinel<TestRequest, TestName> = Sentinel::new(1usize, 1usize);
+        let name = generate_random_name();
+        let request = TestRequest::new(random::<usize>(), name.clone());
+        let claim = TestClaim { value : random::<usize>() };
+        let serialised_claim = claim.serialise();
+        let key_pair = crypto::sign::gen_keypair();
+        let signature = crypto::sign::sign_detached(&serialised_claim, &key_pair.1);
+        let climant_name = generate_random_name();
+        name_key_pairs.push((climant_name.clone(), key_pair.0.clone()));
+
+        // first claim added should return AddResult::RequestKeys
+        match sentinel.add_claim(request.clone(), climant_name.clone(), signature.clone(), serialised_claim.clone()) {
+            Some(result) => match result {
+                AddResult::RequestKeys(source_name) => { assert_eq!(request.get_source(), source_name) },
+                AddResult::Resolved(_, _) => assert!(false)
+                },
+                None => assert!(true)
+            }
+
+        // One key is required should pass
+        match sentinel.add_keys(request.clone(), name_key_pairs.clone()) {
+            Some(result) => { assert_eq!(result.1, serialised_claim);
+                              assert_eq!(result.0, request)},
+            None => assert!(false)
+        }
+    }
+
+#[test]
+    fn request_and_its_duplicate_added() {
+        let mut sentinel: Sentinel<TestRequest, TestName> = Sentinel::new(CLAIM_THRESHOLDS, KEY_THRESHOLDS);
+        let name = generate_random_name();
+        let request = TestRequest::new(random::<usize>(), name.clone());
+        let claim = TestClaim { value : random::<usize>() };
+        let serialised_claim = claim.serialise();
+        let key_pair = crypto::sign::gen_keypair();
+        let signature = crypto::sign::sign_detached(&serialised_claim, &key_pair.1);
+        let climant_name = generate_random_name();
+
+        // first claim added should return AddResult::RequestKeys
+        match sentinel.add_claim(request.clone(), climant_name.clone(), signature.clone(), serialised_claim.clone()) {
+            Some(result) => match result {
+                AddResult::RequestKeys(source_name) => { assert_eq!(request.get_source(), source_name) },
+                AddResult::Resolved(_, _) => assert!(false)
+                },
+                None => assert!(true)
```

Comment:
I agree with peter, please clean up the 'weird usage of asserts', I'll give an example below: